pixleesdk/src/main/java/com/pixlee/pixleesdk/ui/widgets/PXLPhotoProductView.kt

```diff
@@ -170,13 +203,115 @@ class PXLPhotoProductView : FrameLayout, LifecycleObserver {
 
 
 
-                list.layoutManager = LinearLayoutManager(context, LinearLayoutManager.HORIZONTAL, false)
-                list.adapter = adapter
+                recyclerView.layoutManager = LinearLayoutManager(context, LinearLayoutManager.HORIZONTAL, false)
+                recyclerView.adapter = adapter
             }
         }
     }
 
-    var isMutted: Boolean = false
+    private var hiddenHotspots = false
+    private var hotspotsJob: Job? = null
+    private fun addHotspots() {
+        // video does not have hotspots.
+        if (!useHotspots || photoInfo?.pxlPhoto?.isVideo == true) return
+
+        hotspotsJob?.cancel()
+        hotspotsJob = scope.launch {
+            withContext(Dispatchers.IO) {
+                // delay below for other elements like products, and the main content to be loaded with the maximum resources
```

Comment:
Are those views? And if so, do we have access to them here to check whether each one has called their `onCreate`? Also, what happens if they aren't loaded?

Reply:
Are those activities/fragments?
- PXLPhotoProductView extends FrameLayout which is a view and can be imported from Activities and fragments.

do we have access to them here to check whether each one has called their onCreate
- yes we can using LifecycleObserver. Everything of PXLPhotoProductView can be used when or after onCreate of Activity/Fragment is called. The Lifecycle of Activity and Fragment is observed with this code.
```kotlin
#!kotlin
class PXLPhotoProductView : FrameLayout, LifecycleObserver {
    private fun initView(context: Context) {
        ....
        lifecycleOwner.lifecycle.addObserver(this)
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_START)
    fun playVideoOnStart() {
        playVideoOnResume()
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_RESUME)
    fun playVideoOnResume() {
        fireAnalyticsOpenLightbox()
        pxlPhotoView.playVideo()
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_PAUSE)
    fun stopVideoOnPause() {
        pxlPhotoView.pauseVideo()
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_STOP)
    fun stopVideoOnStop() {
        stopVideoOnPause()
    }

    private var isAnalyticsOpenLightboxFired = false

    @OnLifecycleEvent(Lifecycle.Event.ON_RESUME)
    private fun fireAnalyticsOpenLightbox() {
    }
}
```

Also, what happens if they aren't loaded?
- If hotspots aren't loaded, PXLPhotoProductView will work without them. I can't think of any of the cases. I think the only way of not loading hotspots could be that the SDK user manually deletes the parent view(`v_hotspots`) of hotspots.